### ISSUE TITLE
spoof "high accuracy" for whining apps

### ISSUE THREAD

**Issue by TameOfGroans** [microg/android_packages_apps_UnifiedNlp#39]
After one has built a High Precision GNSS local database of mac addresses and mobile towers subsequent selection of "low power" mode should not needlessly inflict notifications urging [or outright requiring] "high power" mode. Glympse, a social gps app, is tedious in this regard. 

Especially for those without g.apps provide an option to spoof power mode reported to location apps /while/ retaining an NLP mode not querying GNSS. 

Can alternate NLP offer more than 3 power modes?

**Comment by Vdragon** [comment]
+1 for this too as Ingress still can't operate normally even when network location base on Google network location provider still could

**Comment by mar-v-in** [comment]
Apps complaining about low accuracy when only NLP and no GPS is active do this for at least one of the following reasons:
1. The location provider is not "gps" - there is no way to come around this issue
2. The app wants to use play services location API instead of system location API - see [this comment](https://github.com/microg/android_packages_apps_UnifiedNlp/issues/36#issuecomment-86744484) on and µg GmsCore for a possible solution
3. The app wants high accuracy and checks the accuracy value accordingly (eg. to be less than 20 meters) - this could be "fixed" by faking a higher accuracy, but shouldn't as this value is intended to have a real meaning. Google's NLP does never use a accuracy better than 42 meters and so does UnifiedNLP

Finally: on my test device running µg GmsCore, I was able to use Ingress without gps being enabled, so either 2 or 3 (or both) apply for it.

**Comment by n76** [comment]
Is there a place where a pre-built µg GmsCore can be downloaded? I don't see it on F-Droid nor at http://forum.xda-developers.com/showthread.php?t=1715375

**Comment by mar-v-in** [comment]
µg GmsCore is unfortunately not fully stable yet and might cause problems like high battery drain or data usage. I won't call it release, not even "beta" grade software yet, that's why I did not publish a release binary yet. It should also be noted that GmsCore does not work with any default or custom rom other than OmniROM 4.4 right now, as [a patch to the system](https://github.com/microg/android_packages_apps_GmsCore/blob/master/android_frameworks_base%2BFAKE_PACKAGE_SIGNATURE.patch) is required.

If you build your ROM yourself or for any reason are capable of using a ROM that integrates the patch, @SWW13 provides nightly builds [here](http://android.brn.li/apps/GmsCore/). Installation can be a little tricky as well and requires the usage of magic dialer codes right now -- as mentioned: it's far away from release grade software...

**Comment by n76** [comment]
Thanks! I guess I've been living without that functionality for a while and I can live without it for a while longer. :)

And thank you so very much for this project and all the work you have done on it!

**Comment by Catfriend1** [comment]
for me , gsf proxy and gms from microg runs out of the box under cm 12.1 a 5.1.1 on device i9100. used the latest build from sww13 debug apks.

**Comment by Catfriend1** [comment]
another issue: playing Ingress with microg gms will get you blocked by niantic after three days of use. you can See this when the Mission tab shows no missions avail error and glyphing is forced.

**Comment by mar-v-in** [comment]
> playing Ingress with microg gms will get you blocked by niantic after three days

I can not reproduce this behavior. Make sure you have a proper location plugin active and gmscore, gsfproxy and a play store client (original play store, blankstore or fake store) installed. Ingress is on the list of apps regularly tested for compatibility with microG.

**Comment by Catfriend1** [comment]
thanks for your answer.
in 01/2015 following config blocked my Ingress account:
- gsf from Google 
- unifiednlp 1.2.0 as gms replacement priv-app.
- Plugins: applewifinlp mozillanlp localwifi nominatim 
- location was accurate, no play store installed. Google login App installed.

i contacted Ingress support, was not unblocked. three tries. two months later i got back to gapps pico package. wrote Ingress support and got unblocked.

now, last weekend switched to gmscore, gsf microg Version sww13 with original playstore 6.0.5. everything fine until now, Paps and licenses ok. no wake problems, no system crash. Ingress block came immediately. i am only partially Blocked, glyphing is enforced on some Portals. i dont have cheating apps. i suspect niantic might have a signature detection. Plugins are installed and the same.

**Comment by mar-v-in** [comment]
Do you have cloud messaging enabled in microG settings?

**Comment by Catfriend1** [comment]
No, GCM is currently disabled in microG settings. I did only enable CheckIn Service. Could that be the reason of the problems?

**Comment by mar-v-in** [comment]
As Ingress makes use of GCM this might be the problem, yes.

**Comment by Catfriend1** [comment]
is it normal that Ingress tells on every startup "Ingress needs Google play services" even if microg is installed and location already Working properly? activating gcm did not solve the Problem. i still cant See missions in ingress.
